pkg/csi/cinder/openstack/openstack_snapshots.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package openstack snapshots provides an implementation of Cinder Snapshot features
+// cinder functions using Gophercloud.
+package openstack
+
+import (
+	"github.com/golang/glog"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/v3/snapshots"
+)
+
+// CreateSnapshot issues a request to take a Snapshot of the specified Volume with the corresponding ID and
+// returns the resultant gophercloud Snapshot Item upon success
+func (os *OpenStack) CreateSnapshot(name, volID, description string, tags *map[string]string) (*snapshots.Snapshot, error) {
+	opts := &snapshots.CreateOpts{
+		VolumeID:    volID,
+		Name:        name,
+		Description: description,
+	}
+	if tags != nil {
+		opts.Metadata = *tags
+	}
+
+	snap, err := snapshots.Create(os.blockstorage, opts).Extract()
+	if err != nil {
+		return &snapshots.Snapshot{}, err
```

Comment:
we could add log print here as well

Reply:
yeah, my opinion was that we probably don't need so much logging in the package, but rather the caller should be grown up and do what needs to be done when it gets an empty Snapshot and an error.  I'm fine either way though and don't mind adding a log message if people prefer it.